crates/astria-conductor/src/reader.rs

```diff
@@ -150,65 +145,72 @@ impl Reader {
     }
 
     /// get_new_blocks fetches any new sequencer blocks from Celestia.
-    pub async fn get_new_blocks(&mut self) -> Result<Vec<SequencerBlockData>> {
-        debug!("ReaderCommand::GetNewBlocks");
-        let mut blocks = vec![];
-
+    #[instrument(name = "Reader::get_new_blocks", skip_all)]
+    pub async fn get_new_blocks(&mut self) -> eyre::Result<Vec<SequencerBlockData>> {
         // get the latest celestia block height
         let prev_height = self.curr_block_height;
-        self.curr_block_height = self.celestia_client.get_latest_height().await?;
+        let curr_block_height = self
+            .celestia_client
+            .get_latest_height()
+            .await
+            .wrap_err("failed getting latest height from celestia")?;
+        ensure!(
+            curr_block_height > self.curr_block_height,
+            "no new celestia height"
+        );
+
         info!(
-            "checking celestia blocks {} to {}",
-            prev_height, self.curr_block_height
+            height.start = prev_height,
+            height.end = curr_block_height,
+            "checking celestia blocks for range of heights",
         );
 
+        let mut blocks = vec![];
         // check for any new sequencer blocks written from the previous to current block height
-        for height in prev_height..self.curr_block_height {
-            let res = self
+        'check_heights: for height in prev_height..=self.curr_block_height {
```

Comment:
what does the `=self.curr_block_height` do?

Reply:
`..=` is to construct an inclusive range, i.e. `[x, y]` rathern than `[x, y)` as for `..`.

The old logic was to check `[old_current_block, new_current_block)` - so we never actually fetched the most up-to-date block.

On that note I just realize that I should be checking `prev_height + 1 ..= curr_block_height`